Find a functionally equivalent code:android.animation.PropertyValuesHolder.setupEndValue:COMMENT
Method Modifier: 
Comment:/**
 * This function is called by ObjectAnimator when setting the end values for an animation.
 * The end values are set according to the current values in the target object. The
 * property whose value is extracted is whatever is specified by the propertyName of this
 * PropertyValuesHolder object.
 *
 * @param target The object which holds the start values that should be set.
 */

@@ -1,3 +1,6 @@
 {
-    setupValue(target, mKeyframeSet.mKeyframes.get(mKeyframeSet.mKeyframes.size() - 1));
+    ArrayList<Keyframe> keyframes = mKeyframes.getKeyframes();
+    if (!keyframes.isEmpty()) {
+        setupValue(target, keyframes.get(keyframes.size() - 1));
+    }
 }

